File: TestUnixDomainSocket.java
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public class TestUnixDomainSocket {

	public static void main(String[] args) throws IOException {
		if (args.length != 1) {
			System.out
					.println("usage: java TestUnixDomainSocket socketfilename");
			System.exit(1);
		}
		String socketFile = args[0];

		byte[] b = new byte[128];
		UnixDomainSocketClient socket = new UnixDomainSocketClient(socketFile,
				UnixDomainSocket.SOCK_STREAM);
		InputStream in = socket.getInputStream();
		OutputStream out = socket.getOutputStream();
		System.out.println("Test #1: Test UnixDomainSocketClient\nTestcase "
				+ "1.1: Test UnixDomainSocketClient with a stream socket...");
		in.read(b);
		System.out.println("Text received: \"" + new String(b) + "\"");
		String text = "Hello! I'm the client!";
		out.write(text.getBytes());
		System.out.println("Text sent: " + "\"" + text + "\"");
		socket.close();

		System.out.println("Testcase 1.2: Test UnixDomainSocketClient with "
				+ "a datagram socket...");
		socket = new UnixDomainSocketClient(socketFile,
				UnixDomainSocket.SOCK_DGRAM);
		System.out.println("Provoke and catch an "
				+ "UnsupportedOperationException:");
		try {
			in = socket.getInputStream();
		} catch (UnsupportedOperationException e) {
			System.out.println("UnsupportedOperationException has been "
					+ "thrown as expected.");
		}
		out = socket.getOutputStream();
		text = "Hello! I'm the client!";
		out.write(text.getBytes());
		System.out.println("Text sent: \"" + text + "\"");
		socket.close();

		UnixDomainSocketServer ssocket = new UnixDomainSocketServer(socketFile,
				UnixDomainSocket.SOCK_STREAM);
		in = ssocket.getInputStream();
		out = ssocket.getOutputStream();
		System.out.println("\nTest #2: Test UnixDomainSocketServer\nTestcase "
				+ "2.1: Test UnixDomainSocketServer with a stream socket...");
		in.read(b);
		System.out.println("Text received: \"" + new String(b) + "\"");
		text = "Hello! I'm the server!";
		out.write(text.getBytes());
		System.out.println("Text sent: " + "\"" + text + "\"");
		ssocket.close();
		ssocket.unlink();

		System.out.println("Testcase 2.2: Test UnixDomainSocketServer with "
				+ "a datagram socket...");
		ssocket = new UnixDomainSocketServer(socketFile,
				UnixDomainSocket.SOCK_DGRAM);
		System.out.println("Provoke and catch an "
				+ "UnsupportedOperationException:");
		in = ssocket.getInputStream();
		try {
			out = ssocket.getOutputStream();
		} catch (UnsupportedOperationException e) {
			System.out.println("UnsupportedOperationException has been "
					+ "thrown as expected.");
		}
		in.read(b);
		System.out.println("Text received: \"" + new String(b) + "\"");
		ssocket.close();
		ssocket.unlink();
	}
}

File: Test_UnixDomainSocket.py
#!/usr/bin/env python

import sys, os, socket, time

BUFSIZE = 128

def main():
    if len(sys.argv) != 2:
        print "usage: %s socketfilename" % sys.argv[0]
        sys.exit(1)
    socket_file = sys.argv[1]

    print ("Test #1: Test UnixDomainSocketClient\n"
        "Testcase 1.1: Test UnixDomainSocketClient with a stream socket...")
    s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    s.bind(socket_file)
    s.listen(0)
    client, client_address = s.accept()
    text = "Hello I'm the server..."
    client.send(text)
    print "Text sent: \"Hello I'm the server...\""
    text = client.recv(BUFSIZE)
    print "Text received: \"%s\"" % text
    s.close()
    os.unlink(socket_file)

    print "Testcase 1.2: Test UnixDomainSocketClient with a datagram socket..."
    s = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
    s.bind(socket_file)
    text = s.recv(BUFSIZE)
    print "Text received: \"%s\"" % text
    s.close()
    os.unlink(socket_file)

    print ("\nTest #2: Test UnixDomainSocketServer\n"
        "Testcase 2.1: Test UnixDomainSocketServer with a stream socket...")
    s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
    time.sleep(2)   # wait for the server
    s.connect(socket_file)
    text = "Hello I'm the client..."
    s.send(text)
    print "Text sent: \"Hello I'm the client...\""
    text = s.recv(BUFSIZE)
    print "Text received: \"%s\"" % text
    s.close()

    print "Testcase 2.2: Test UnixDomainSocketServer with a datagram socket..."
    s = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
    time.sleep(2)   # wait for the server
    s.connect(socket_file)
    text = "Hello I'm the client..."
    s.send(text)
    print "Text sent: \"%s\"" % text
    s.close()


if __name__ == "__main__":
    main()
